### ISSUE TITLE
Libmesh linking fails with address sanitizer enabled

### ISSUE THREAD

**Issue by dschwen** [libMesh/libmesh#1396]
Build fails with

```
Undefined symbols for architecture x86_64: "___asan_after_dynamic_init", referenced from:
...
```

with

```
--enable-sanitize=devel
```

when using the latest Xcode (8.3.3). It worked with at least 8.1

**Comment by jwpeterson** [comment]
Can you come up with a simple test code that fails? We test the sanitize compiler flags with a super simple main program in m4/sanitize.m4, but presumably this passes on your system, while using sanitizer flags on "real" codes fails to link for some reason.

**Comment by dschwen** [comment]
Spent 30mins on it. No success. Hard to figure out what exactly triggers it, as it fails on linking, not on compile.

**Comment by dschwen** [comment]
Ok, got it:

Makefile:
```
test: test.o
	clang++ test.o -o test -L/opt/moose/llvm-3.9.0/lib/clang/3.9.0/lib/darwin -lclang_rt.osx

test.o: test.cc
	clang++ -std=gnu++11 -march=native -fsanitize=address test.cc -c -o test.o
```

test.cc:
```
#include <vector>
#include <iostream>

typedef unsigned long dof_id_type;
typedef dof_id_type index_t;
typedef std::vector<index_t> index_buffer_t;
index_buffer_t _idx_buf;

int main()
{
	index_buffer_t().swap(_idx_buf);
	return _idx_buf.empty() ? 0 : 1;
}
```

**Comment by dschwen** [comment]
The `-lclang_rt.osx` is key

**Comment by jwpeterson** [comment]
What if you take out that`-march=native` flag?

I think you are the only person using this... maybe it has a bad interaction with the address sanitizer.

**Comment by jwpeterson** [comment]
I ask because I don't think we have control over `-lclang_rt.osx`, it must be something coming from libtool?

**Comment by dschwen** [comment]
Same deal ` -march=native` does not matter

**Comment by jwpeterson** [comment]
I believe I have traced this issue to needing to pass the `-fsanitize=address` flag to the linker as well as the compiler when compilation is done in separate stages. Doing so fixes the example posted above, anyway. I think the fix will be to ensure that the sanitize flag gets added to the `LDFLAGS` variable during configure...

**Comment by jwpeterson** [comment]
Fixed by #1406.
